### PR TITLE
Add `resultType` and `resultTypeUntyped` methods to MethodBuilder

### PR DESCRIPTION
**Pull request by elliottt**
Add methods to `MethodBuilder` to support setting the `resultType` of the symbol being defined. `resultTypeUntyped` is a special case, as it requires the `MethodRef` of the method being defined.

### Motivation

More consistent api for `MethodBuilder`.

### Test plan

See included automated tests.

### CODE REVIEW

File: core/GlobalState.cc [hunk 1]
```diff
@@ -110,6 +110,16 @@ struct MethodBuilder {
...
 
+    MethodBuilder &resultTypeUntyped() {
+        method.data(gs)->resultType = Types::untyped(gs, method);
+        return *this;
+    }
+
+    MethodBuilder &resultType(TypePtr &&type) {
+        method.data(gs)->resultType = type;
+        return *this;
+    }
```
Comment (froydnj) [inline core/GlobalState.cc:121]: Could these be done as `buildWithUntypedResult` and `buildWithResult(TypePtr &&type)` and thus replace `build()`?

Comment (elliottt) [reply core/GlobalState.cc:121]: Seems reasonable to me!

Comment (elliottt) [reply core/GlobalState.cc]: I couldn't remove `build` completely, but I did make these two functions return the `MethodRef` instead.

**Review by aprocter-stripe — APPROVED** [review]
I think I actually like `resultType` a little better than `buildType`, unless we need to force setting of the result type to be the last step for some safety reason I don't know about. Don't have strong feelings, though, and things are definitely cleaner after this change than they were before it.

**Comment by elliottt** [comment]
> I think I actually like `resultType` a little better than `buildType`, unless we need to force setting of the result type to be the last step for some safety reason I don't know about. Don't have strong feelings, though, and things are definitely cleaner after this change than they were before it.

I like it too, but this does make it harder to miss setting it, and also makes it so that you can't overwrite it without having to jump through a lot of hoops.

**Pull request merged**